### PR TITLE
Hyundai: Car Port for Ioniq 5 (HDA2) 2022

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: `22de8111a8c5463c|2022-07-29--13-34-49`
- [x] harness type: [Hyundai P](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) ~[Hyundai K](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) (modified)~
  - ~HDA2 specific: User had to modify the harness to flip the can bus pairs in order to successfully intercept the `LKAS (0x50)` steering message~

**Physical setup**
- Top port behind comma three
  - comma Power <-> First Harness Box <-> comma three USB C/Right-Angle C <-> comma three
- Bottom port behind comma three
  - Camera Harness <-> Second Harness Box <-> USB C/C 3.1 <-> Red Panda <-> USB A/A <-> USB A to C **OTG Adapter** <-> comma three

Thanks to community Ioniq 5 owner spectre3ooo and G80Tampa.

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -131,6 +132,7 @@ class HyundaiCarInfo(CarInfo):
   ],
   CAR.VELOSTER: HyundaiCarInfo("Hyundai Veloster 2019-20", min_enable_speed=5. * CV.MPH_TO_MS, harness=Harness.hyundai_e),
   CAR.SONATA_HYBRID: HyundaiCarInfo("Hyundai Sonata Hybrid 2020-22", "All", harness=Harness.hyundai_a),
+  CAR.IONIQ_5: HyundaiCarInfo("Hyundai Ioniq 5 2022", "All", harness=Harness.hyundai_k),
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:135]: is the harness modified or stock k works?

Comment (adeebshihadeh) [reply selfdrive/car/hyundai/values.py:135]: can you make this Harness.none for now?

Comment (vanillagorillaa) [reply selfdrive/car/hyundai/values.py:135]: > is the harness modified or stock k works? @vanillagorillaa looks like we'll need another hyundai harness

What a joy, another one!

Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:135]: let's do this for now
```suggestion
  CAR.IONIQ_5: HyundaiCarInfo("Hyundai Ioniq 5 2022", "HDA2", harness=Harness.none),
```

**Comment by adeebshihadeh** [comment]
Do disengage on gos and resume spam both work? I know that LFA isn't properly blocked on the Ioniq 5. I would first try sending zero's for all the bytes in CAM_0x2A4. https://github.com/commaai/openpilot/blob/22408781f7d7df5ff0395f80c1b15d6407acd0ff/selfdrive/car/hyundai/hda2can.py#L16

**Comment by sunnyhaibin** [comment]
> Do disengage on gos and resume spam both work? I know that LFA isn't properly blocked on the Ioniq 5. I would first try sending zero's for all the bytes in CAM_0x2A4.
> 
> https://github.com/commaai/openpilot/blob/22408781f7d7df5ff0395f80c1b15d6407acd0ff/selfdrive/car/hyundai/hda2can.py#L16

User mentioned disengage on accelerator sometimes would work but inconsistent.

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -126,6 +127,7 @@ class HyundaiCarInfo(CarInfo):
   ],
   CAR.VELOSTER: HyundaiCarInfo("Hyundai Veloster 2019-20", "Smart Cruise Control (SCC)", min_enable_speed=5. * CV.MPH_TO_MS, harness=Harness.hyundai_e),
   CAR.SONATA_HYBRID: HyundaiCarInfo("Hyundai Sonata Hybrid 2020-22", "All", harness=Harness.hyundai_a),
+  CAR.IONIQ_5: HyundaiCarInfo("Hyundai Ioniq 5 2022", "HDA2", harness=Harness.none),
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:130]:
```suggestion
  CAR.IONIQ_5: HyundaiCarInfo("Hyundai Ioniq 5 2022", "Highway Driving Assist 2", harness=Harness.none),
```

Let's use the full market name, as this is now shown on the website.

**Comment by adeebshihadeh** [comment]
> User mentioned disengage on accelerator sometimes would work but inconsistent.

Discussed with our two uses on Discord, and they're reporting both gas disengage and LFA blocking work as expected now.

**Comment by adeebshihadeh** [comment]
Congrats on #200 @sunnyhaibin!

**Pull request merged**